Sources/PackageModelSyntax/AddTarget.swift

```diff
@@ -57,6 +57,71 @@ public enum AddTarget {
         }
     }
 
+    // Check if the package has a single target with that target's sources located
+    // directly in `./Sources`. If so, move the sources into a folder named after
+    // the target before adding a new target.
+    package static func moveSingleTargetSources(
```

Comment:
suggestion (possibly-blocking): can we write automated test that validate this function in isolation, where we can test various code paths?